sous-chef/sous_chef/grocery_list/generate_grocery_list/make_grocery_list.py

```diff
@@ -0,0 +1,171 @@
+        # ------ error for that recipe that will be added to the exception
+        # ------ handler so other recipes can be continued. add ingredients to
+        # ------ dataframe. with each evaluation of a sub-recipe, we are asked
+        # ------ about preparing this, for simplicity, basically like
+        # ------ _add_referenced_recipe_to_queue but cleaner ---would be nice
+        # ------ if modeled similar to _process_menu_recipe's usage
+        # ------ in _fill_menu_template
+        # ------ (later step) keep track of already called sub-recipes to
+        # ------ reduce re-asking questions & keep track of total amount needed
+        # --- later concatenate/process ingredients
+        # --- outputs a non-aggregated dataframe -> just ingredients, dates
+        # --- outputs a prep list that would go to todoist (this should
+        # --- NOT have task formatting -> instead do in todoist function
+        # send prep list to todoist, if exists & in prod mode
+        #
+        #
+        # --- functions (not exhaustive) now obsolete and to be removed from old
+        # --- & their tests moved to the new functions
+        # ---- _add_menu_recipe_to_queue
+        # ---- _add_bulk_manual_ingredient_to_grocery_list
+        # ---- _process_recipe_queue
+
+        menu = Menu(config=self.config)
+        (
+            menu_ingredient_list,
+            menu_recipe_list,
+        ) = menu.get_menu_for_grocery_list()
+
+        grocery_list = GroceryListOld(
+            self.config.grocery_list,
+            due_date_formatter=self.due_date_formatter,
+            ingredient_field=self._get_ingredient_field(),
+            unit_formatter=self.unit_formatter,
+        )
+        return grocery_list.extract_ingredients_from_menu(
+            menu_ingredient_list, menu_recipe_list
+        )
+
+    def prepare_grocery_list(
+        self, raw_grocery_df: pd.DataFrame
+    ) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        # use the output from extract_ingredients_from_menu
+        # use new class, i.e. GroceryListAggregator
+        # and its function, aggregate_ingredients
+        # --- this function will call on functions contained in its class
+        # --- that are in _aggregate_grocery_list
+        # --- outputs an aggregrated grocery list in a dataframe
+
+        # send aggregated grocery list to todoist, if prod mode
+        # todoist class does formatting of strings
+
+        grocery_list = GroceryListOld(
+            self.config.grocery_list,
+            due_date_formatter=self.due_date_formatter,
+            ingredient_field=self._get_ingredient_field(),
+            unit_formatter=self.unit_formatter,
+        )
+
+        return grocery_list.prepare_grocery_list(raw_grocery_df=raw_grocery_df)
+
+    def export_to_todoist(
+        self, grocery_list_df: pd.DataFrame, prep_task_df: pd.DataFrame
+    ) -> None:
+        # switch to class specific to todoist output
```

Comment:
This was already started some in `GroceryListForTodoist`, but then not continued & there is some duplicate code too. The main difficulty with continuing it is that there are misc. formatting functions for strings (task & beans) that are mixed in to various places. It would be doable, but maybe a bit more effort + needing to mind testing it properly as we're moving it about.